controllers/vspherevm_controller.go

```diff
@@ -306,6 +308,31 @@ func (r vmReconciler) reconcileDelete(ctx *context.VMContext) (reconcile.Result,
 		r.Logger.V(6).Info("unable to delete node", "err", err)
 	}
 
+	// Remove finalizers from any ipam claims
```

Comment:
If this controller is responsible for querying and removing the finalizer on the `IPAdressClaim`, then the kubebuilder brace directives should be a part of this file as well. Any particular reason why these get set on the `vspheremachine_controller.go` only.
My vote would be to duplicate for code readability if it is needed in both places. That way each controller clearly states all the rbac permissions it requires.

Reply:
Good catch, this was a mistake. When we started this work we assumed we'd be modifying the `vspheremachine_controller.go`, but it turned out we needed to modify the `vspherevm_controller.go` instead. We've moved the kubebuilder brace directives to `vspherevm_controller.go`.